ZenPacks/zenoss/Import4/service_definition/imp4mariadb.json

```diff
@@ -143,14 +165,14 @@
        "Actions": null,
        "HealthChecks": {
          "running": {
-           "Script": "mysql -uroot -hlocalhost -P3306 -e 'select 1' > /dev/null",
+           "Script": "mysql --socket=/var/lib/mysql/mysql.sock -uroot -hlocalhost -P3306 -e 'select 1' && mysql --socket=/var/lib/mysql.model/mysql.sock -uroot -hlocalhost -P3307 -e 'select 1' > /dev/null",
            "Interval": 5.0
          }
        },
        "Prereqs": [
          {
-           "Name": "MariaDB Availability",
-           "Script": "su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --usedb zodb --execsql=\";\"'"
+           "Name": "MariaDB zodb and zep Availability",
+           "Script": "su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --dbport 3306 --usedb zep --execsql=\";\"' && su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --dbport 3307 --usedb zodb --execsql=\";\"' "
```

Comment:
For now, the implementation is assuming (which may not be correct) that
both mariadb backends run on the same host. The imp4mariadb will be another
container running on the same host. Because of the GRANT problem to
manipulate the tables, the imp4mariadb must run on the same host using -u
root without a password.

We did specify the host:port by -hlocalhost -P3306/3307 for mysql command.

The --socket is a file which is mounted as a volume. Two backends using two
different socket.
We retain the default ones for events db and use a new port/socket for zodb.
The host is default to localhost.

Thanks!


On Wed, Mar 25, 2015 at 10:29 AM, Andrew Lapidas <notifications@github.com>
wrote:

> In ZenPacks/zenoss/Import4/service_definition/imp4mariadb.json
> <https://github.com/zenoss/ZenPacks.zenoss.Import4/pull/16#discussion_r27131744>
> :
>
> >             "Interval": 5.0
> >           }
> >         },
> >         "Prereqs": [
> >           {
> > -           "Name": "MariaDB Availability",
> > -           "Script": "su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --usedb zodb --execsql=\";\"'"
> > +           "Name": "MariaDB zodb and zep Availability",
> > +           "Script": "su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --dbport 3306 --usedb zep --execsql=\";\"' && su - zenoss -c '/opt/zenoss/bin/python /opt/zenoss/Products/ZenUtils/ZenDB.py --dbport 3307 --usedb zodb --execsql=\";\"' "
>
> Ditto on --socket comment
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zenoss/ZenPacks.zenoss.Import4/pull/16/files#r27131744>
> .
>



-- 
Thanks,
----
Alex